projects/packages/my-jetpack/src/class-initializer.php

```diff
@@ -211,6 +224,64 @@ public static function enqueue_scripts() {
 		}
 	}
 
+	/**
+	 * Determine if the current user is "new" to Jetpack
+	 * This is used to vary some messaging in My Jetpack
+	 *
+	 * On the front-end, purchases are also taken into account
+	 *
+	 * @return bool
+	 */
+	public static function is_jetpack_user_new() {
+		// is the user connected?
+		$connection = new Connection_Manager();
+		if ( $connection->is_user_connected() ) {
+			return false;
+		}
+
+		// Check to see if there's been a heartbeat recorded
+		// This would tell us that the user has been connected at least once in the past,
+		// even if they are disconnected now
+		$last_heartbeat = (int) \Jetpack_Options::get_option( 'last_heartbeat' );
+		if ( $last_heartbeat && $last_heartbeat > 0 ) {
+			return false;
+		}
+
+		// are any modules active?
+		$modules        = new Modules();
+		$active_modules = $modules->get_active();
+		// if the Jetpack plugin is active, filter out the modules that are active by default
+		if ( class_exists( 'Jetpack' ) && ! empty( $active_modules ) ) {
+			$active_modules = array_diff( $active_modules, Jetpack::get_default_modules() );
+		}
+		if ( ! empty( $active_modules ) ) {
+			return false;
+		}
+
+		// check for other Jetpack plugins that are installed on the site (active or not)
+		// If there's more than one Jetpack plugin active, this user is not "new"
+		$plugin_slugs              = array_keys( Plugins_Installer::get_plugins() );
+		$plugin_slugs              = array_map(
+			static function ( $slug ) {
+				$parts = explode( '/', $slug );
+				return str_replace( '.php', '', $parts[ is_countable( $parts ) ? count( $parts ) - 1 : 0 ] );
```

Comment:
`explode()` will always return an array, that is countable 🤔 If `count( $parts ) is 1`, then `count( $parts ) - 1 === 0`. If `count( $parts ) is 0`, accessing element `$parts[0]` will also throw an error.

Reply:
> explode() will always return an array, that is countable 🤔
Good note, I've become so used to adding `is_countable` to avoid the PHP 8.1 warning 😅 

>  If count( $parts ) is 0, accessing element $parts[0] will also throw an error
Good thought, I can add an empty check.